Library/ViewModels/PaymentSourceSelected.swift

```diff
@@ -0,0 +1,21 @@
+import Foundation
+public enum PaymentSourceSelected: Equatable {
+  case paymentSourceId(String)
+  case setupIntentClientSecret(String)
+
+  public var paymentSourceId: String? {
+    if case let .paymentSourceId(value) = self {
+      return value
+    } else {
+      return nil
+    }
+  }
+
+  public var setupIntentClientSecret: String? {
+    if case let .setupIntentClientSecret(value) = self {
+      return value
+    } else {
+      return nil
+    }
+  }
+}
```

Comment:
any particular reason you didn't want to use switch statements here?

Reply:
Hm, not really. Just tried it out and it's a little neater looking, so I'll swap these.